uint8 dimension # dimension of state 2 for 2d 3 for 3d
float64[] vars # pos, vel, acc, ...